forc-util/src/lib.rs

```diff
@@ -64,6 +64,26 @@ pub fn format_log_receipts(receipts: &[fuel_tx::Receipt], pretty_print: bool) ->
     }
 }
 
+/// Continually go down in the file tree until a Forc manifest file is found.
+///
+/// Given a pacakge's directory (which contains a manifest file) goes down in the file tree until
+/// another manifest file is found.
+///
+/// Returns the directory that contains the nested manifest file.
+pub fn find_nested_manifest_dir(starter_path: &Path) -> Option<PathBuf> {
+    use walkdir::WalkDir;
+    WalkDir::new(starter_path)
+        .into_iter()
+        .filter_map(|e| e.ok())
+        .filter(|entry| entry.file_name().to_string_lossy() == constants::MANIFEST_FILE_NAME)
+        .map(|entry| {
+            let mut entry = entry.path().to_path_buf();
+            entry.pop();
+            entry
+        })
+        .nth(1)
```

Comment:
I'm not sure that `.nth(1)` is what we want here, as I think it assumes `starter_path` will always contain a manifest file :thinking: This might break in the case that this function is used to search for a nested manifest directory starting from a directory that does not already contain a manifest. I realize the doc comment mentions this, but perhaps we can make this function a little more intuitive/flexible by removing this constraint?

Instead, perhaps we should use a filter or something to check that the `entry.path()`'s directory is not equal to the starter path directory?